src/Host/Broker/Impl/Sessions/Session.cs

```diff
@@ -74,7 +77,7 @@ public class Session {
             _pipe = new MessagePipe(messageLogger);
         }
 
-        public void StartHost(SecureString password, string profilePath, ILogger outputLogger, LogVerbosity verbosity) {
+        public void StartHost(ClaimsPrincipal cp, string profilePath, ILogger outputLogger, LogVerbosity verbosity) {
```

Comment:
The only reason it is here is to reduce lifetime of the plain text password in

cp.FindFirst(Claims.Password)?.Value.ToSecureString()




---
In reply to: [82294590](https://github.com/Microsoft/RTVS/pull/2447#discussion_r82294590) [](ancestors = 82294590)